packages/assertion-monitor/index.ts

```diff
@@ -0,0 +1,200 @@
+    enableAlerting: { type: 'boolean', default: false },
+  })
+  .strict()
+  .parseSync()
+
+const configFileContent = fs.readFileSync(
+  path.join(process.cwd(), options.configPath),
+  'utf-8'
+)
+const config = JSON.parse(configFileContent)
+
+if (!Array.isArray(config.childChains) || config.childChains.length === 0) {
+  console.error('Error: Chains not found in the config file.')
+  process.exit(1)
+}
+
+type ChunkProcessFunction<T> = (
+  fromBlock: bigint,
+  toBlock: bigint,
+  client: PublicClient
+) => Promise<T>
+// Updated processChunkedRange function
+async function processChunkedRange<T>(
+  fromBlock: bigint,
+  toBlock: bigint,
+  chunkSize: bigint,
+  client: PublicClient,
+  processChunk: ChunkProcessFunction<T>
+): Promise<T[]> {
+  const results: T[] = []
+
+  if (fromBlock === toBlock) {
+    const result = await processChunk(fromBlock, toBlock, client)
+    results.push(result)
+    return results
+  }
+
+  let currentFromBlock = fromBlock
+
+  while (currentFromBlock <= toBlock) {
+    const currentToBlock =
+      currentFromBlock + chunkSize - 1n < toBlock
+        ? currentFromBlock + chunkSize - 1n
+        : toBlock
+
+    const result = await processChunk(currentFromBlock, currentToBlock, client)
+    results.push(result)
+
+    if (currentToBlock === toBlock) break
+
+    currentFromBlock = currentToBlock + 1n
+  }
+
+  return results
+}
+const getBlockRange = async (
+  client: PublicClient,
+  childChainInfo: ChainInfo
+) => {
+  const latestBlockNumber = await client.getBlockNumber()
+  const toBlock = BigInt(options.toBlock) || latestBlockNumber
+
+  if (options.fromBlock) {
+    return { fromBlock: BigInt(options.fromBlock), toBlock }
+  }
+
+  const blockTime = getDefaultBlockRange(
+    getChainFromId(childChainInfo.parentChainId)
+  )
+
+  const fromBlock = await client.getBlock({
+    blockNumber: latestBlockNumber - BigInt(blockTime),
+  })
+
+  return { fromBlock: fromBlock.number, toBlock }
+}
+
+const monitorNodeCreatedEvents = async (childChainInfo: ChainInfo) => {
+  const parentChain = getChainFromId(childChainInfo.parentChainId)
+  const client = createPublicClient({
+    chain: parentChain,
+    transport: http(childChainInfo.parentRpcUrl),
+  })
+
+  const { fromBlock, toBlock } = await getBlockRange(client, childChainInfo)
+
+  const getLogsForChunk = async (
+    chunkFromBlock: bigint,
+    chunkToBlock: bigint,
+    client: PublicClient
+  ) => {
+    return client
+      .getLogs({
+        address: childChainInfo.ethBridge.rollup as `0x${string}`,
+        event: nodeCreatedEventAbi,
+        fromBlock: chunkFromBlock,
+        toBlock: chunkToBlock,
+      })
+      .catch(console.error)
+  }
+
+  const logsArray = await processChunkedRange(
+    fromBlock,
+    toBlock,
+    CHUNK_SIZE,
+    client,
+    getLogsForChunk
+  )
+
+  const logs = logsArray.flat()
+
+  const isDefaultSettings = !options.fromBlock && !options.toBlock
```

Comment:
I think allowing for a `fromBlock` and a `toBlock` complicates things a bit more. I don't see why we would need to query a range of blocks that's not the latest one (since this is monitoring current assertion posting). And also, querying a block range in the past might lead to unexpected behavior since we won't be able to find the exact batch that contains the information of the potential assertion created.

Reply:
I think this is not valid anymore (getting values from `options.fromBlock`. 

Also, below, I think it makes sense to keep `from block ... to block ...` or calculate the number of days from the DEFAULT_TIMESPAN constant, to avoid having to modify the text if we end up modifying the constant value.